docker/Dockerfile

```diff
@@ -31,48 +30,28 @@ RUN yum -y install curl wget bzip2 make git patch unzip bison yasm autoconf auto
 ENV PATH /opt/rh/devtoolset-2/root/usr/bin:$PATH
 ENV LD_LIBRARY_PATH /opt/rh/devtoolset-2/root/usr/lib64:/opt/rh/devtoolset-2/root/usr/lib
 
+# Shell scripts for build
+RUN git clone https://github.com/matthew-brett/manylinux.git -b refactor-dockerfile
 
 # Compile the latest Python releases.
 # (In order to have a proper SSL module, Python is compiled against openssl-1.0.2e
 # which is linked statically, and then we delete openssl afterwards)
 
-RUN export OPENSSL_VERSION=openssl-1.0.2f && \
-    export OPENSSL_SHA256=932b4ee4def2b434f85435d9e3e19ca8ba99ce9a065a61524b429a9d5e9b2e9c && \
-    echo "${OPENSSL_SHA256}  ${OPENSSL_VERSION}.tar.gz" > ${OPENSSL_VERSION}.tar.gz.sha256 && \
-    wget http://www.openssl.org/source/${OPENSSL_VERSION}.tar.gz && \
-    sha256sum -c ${OPENSSL_VERSION}.tar.gz.sha256 && \
-    tar -xzf ${OPENSSL_VERSION}.tar.gz && \
-    (cd ${OPENSSL_VERSION} && ./config no-ssl2 no-shared -fPIC --prefix=/usr/local/ssl && make && make install) && \
-    rm -rf ${OPENSSL_VERSION} ${OPENSSL_VERSION}.tar.gz ${OPENSSL_VERSION}.tar.gz.sha256 && \
-    for PYVER in ${PY_VERS}; do  \
-        wget -q https://www.python.org/ftp/python/$PYVER/Python-$PYVER.tgz && \
-        tar -xzf Python-$PYVER.tgz && \
-        (cd Python-$PYVER && \
-            mkdir -p /opt/$PYVER/lib && \
-            LDFLAGS="-Wl,-rpath /opt/$PYVER/lib" ./configure --prefix=/opt/$PYVER --enable-shared --with-ensurepip=install && \
-            make -j2 && \
-            make install \
-        ) && \
-        if [[ "$(echo $PYVER | cut -d. -f 1)" == "3" ]]; then \
-            ln -s /opt/$PYVER/bin/python3 /opt/$PYVER/bin/python; \
-            ln -s /opt/$PYVER/bin/pip3 /opt/$PYVER/bin/pip; \
-        fi; \
-        ln -s /opt/$PYVER/ /opt/$(echo $PYVER | cut -d. -f 1,2); \
-        /opt/$PYVER/bin/pip install pip --upgrade; \
-        /opt/$PYVER/bin/pip install wheel; \
-        rm -rf Python-$PYVER.tgz Python-$PYVER; \
-    done && \
-    rm -rf /usr/local/ssl
-
+RUN ./manylinux/docker/build_openssl.sh openssl-1.0.2f 932b4ee4def2b434f85435d9e3e19ca8ba99ce9a065a61524b429a9d5e9b2e9c
+RUN curl -LO https://bootstrap.pypa.io/get-pip.py
+RUN for py_ver in $PY_VERS; do \
+    ./manylinux/docker/build_python.sh $py_ver; \
+    done
+RUN rm -rf /usr/local/ssl get-pip.py
 
 # Install patchelf and auditwheel (latest)
 RUN wget http://nixos.org/releases/patchelf/patchelf-0.8/patchelf-0.8.tar.gz && \
     tar -xzvf patchelf-0.8.tar.gz && \
     (cd patchelf-0.8 && ./configure && make && make install) && \
-    rm -rf patchelf-0.8.tar.gz patchelf-0.8 && \
-    /opt/3.5/bin/pip install git+git://github.com/manylinux/auditwheel.git && \
-    ln -s /opt/3.5/bin/auditwheel /usr/local/bin/auditwheel
+    rm -rf patchelf-0.8.tar.gz patchelf-0.8
 
+RUN /opt/3.5/bin/pip install git+git://github.com/manylinux/auditwheel.git && \
+    ln -s /opt/3.5/bin/auditwheel /usr/local/bin/auditwheel
```

Comment:
Unrelatedly, @rmcgibbo: maybe we should be switching to the pypi version of auditwheel?

Yes, probably. I'll make a pypi release of the latest HEAD.

> Possibly installed as /opt/3.5/bin/pip install auditwheel==1.0.0, and then updating that at each release as a cache-buster?

I'm not really sure what this means.